src/pages/pt-BR/guides/typescript.md

```diff
@@ -0,0 +1,86 @@
+---
+layout: ~/layouts/MainLayout.astro
+title: TypeScript
+description: Aprenda como utilizar o suporte integrado a TypeScript do Astro.
+i18nReady: true
+---
+
+Astro vem com suporte integrado para [TypeScript](https://www.typescriptlang.org/). Você pode importar arquivos `.ts` e `.tsx` em seu projeto Astro, e até mesmo escrever código TypeScript dentro de seu [componente Astro](/pt-BR/core-concepts/astro-components/#script-do-componente).
+
+O Astro em si não realiza checagem de tipo. A checagem de tipo deve ser realizada fora do Astro, seja pela sua IDE ou por um script separado. A [extensão para VSCode do Astro](/pt-BR/editor-setup/) automaticamente providencia dicas e erros do TypeScript em seus arquivos abertos.
+
+
+## Configuração
+
+É **altamente recomendado** que você crie um arquivo `tsconfig.json` em seu projeto, para que ferramentas como Astro e o VSCode saibam como entender o seu projeto. Algumas funcionalidades (como importação de pacotes do npm) não são completamente suportadas no TypeScript sem um arquivo `tsconfig.json`. 
+
+Algumas opções de configuração do TypeScript precisam de atenção especial no Astro. Abaixo está nosso arquivo `tsconfig.json` inicial recomendado, que você pode copiar e colar em seu próprio projeto. Cada [template em astro.new](https://astro.new/) inclui este arquivo `tsconfig.json` por padrão.
+
+```json
```

Comment:
```suggestion
```js
```
It is not related to your PR but I'd suggest to astro time to change that or to remove comments here!

Reply:
Does something sounds off about comments inside json code blocks? I'm not sure about what's wrong here. But since this is related to the original docs it would be better as its own issue/PR.